src/Sprites/spriteMap.ts

```diff
@@ -0,0 +1,627 @@
+import { Engine } from "../Engines/engine";
+import { IDisposable, Scene } from "../scene";
+import { Nullable } from "../types";
+import { Vector2, Vector3 } from "../Maths/math.vector";
+//import { Color3 } from "../Maths/math.color";
+//import { Camera } from "../Cameras/camera";
+import { Texture } from "../Materials/Textures/texture";
+import { RawTexture } from "../Materials/Textures/rawTexture";
+import { ShaderMaterial } from "../Materials/shaderMaterial";
+import { Mesh } from "../Meshes/mesh";
+import { Observer } from "../Misc/observable";
+import { ExecuteCodeAction } from "../Actions/directActions";
+import { ActionManager } from "../Actions/actionManager";
+import { PickingInfo } from "../Collisions/pickingInfo";
+
+import "../Meshes/Builders/planeBuilder";
+
+/**
+ * Defines the minimum interface to fulfill in order to be a sprite map.
+ */
+export interface ISpriteMap extends IDisposable {
+
+}
+
+/**
+ * Class used to manage a grid restricted sprite deployment on an Output plane.
+ */
+export class SpriteMap implements ISpriteMap {
+
+    /** The Name of the spriteMap */
+    public name : string;
+
+    /** The JSON file with the frame and meta data */
+    public atlasJSON : {frames: Array<any>, meta: object};
+
+     /** The systems Sprite Sheet Texture */
+    public spriteSheet : Texture;
+
+    /** Arguments passed with the Constructor */
+    public options : any;
+
+    /** Public Sprite Storage array, parsed from atlasJSON */
+    public sprites : Array<any>;

```

Comment:
I would rather prefer having an interface for each any. It will be better for type analysis